### PR TITLE
Switch to a loop and not changing directories

### PR DESCRIPTION
**Pull request by ausmith**

### CODE REVIEW

File: gs.sh [hunk 1]
```diff
@@ -1,22 +1,14 @@
...
+DIRS="consumer-layouts
+consumer-domain
+consumer-session
+domain-values
+consumer-main"
+
+for dir in $DIRS
+do
+  echo "$(tput setaf 5)$dir status:$(tput sgr0)"
+  git --git-dir=$dir/.git --work-tree=$dir status
```
Comment (oochelz) [inline gs.sh:12]: Mind giving me a quick explanation of this, Aaron? I want to learn :)
Also, thanks for making this half its original size!!

Comment (ausmith) [inline gs.sh:12]: `--git-dir`: indicates what git directory to reference for this run
`--work-tree`: indicates where to use the updates it gets from git

So one run of the loop would be effectively doing:
`git --git-dir=consumer-layouts/.git --work-tree=consumer-layouts status`
Which basically comes out to "update my consumer-layouts directory using its git references found in consumer-layouts/.git".

That make sense?

**Comment by oochelz** [comment]
I feel like this is much faster, too!

File: gs.sh (hunk 1 above)
Comment (oochelz) [inline gs.sh:12]: How can I apply this to pbp so that it doesn't need to be like so:

``` sh
cd consumer-layouts
...
cd ./consumer-main
```

**Comment by ausmith** [comment]
I actually run the following commands when I want to update the world to the latest on master:

``` sh
$ cd dev
$ find . -type d -depth 2 -name '.git' -exec git --git-dir={} --work-tree=$PWD/{}/.. checkout master \;
$ find . -type d -depth 2 -name '.git' -exec git --git-dir={} --work-tree=$PWD/{}/.. pull origin master \;
```

The first will checkout the master branch (in case I have another branch checked out at the time). The second will pull the latest for `origin/master`. The find command is doing the following:

`find .` indicates starting from the current directory.
`-type d` says look for directories.
`-depth 2` means don't go more than 2 folders deep.
`-name '.git'` is the regex to find instances of the `.git` directory so we only run against those that are git repos.
`-exec git --git-dir={} --work-tree=$PWD/{}/.. checkout master \;` is "run git on each element that `find` finds." `{}` indicates the element that was found and `$PWD` is the present working directory (possibly not needed). Same `--work-tree` and `--git-dir` explanation as before.

Sorry if over explained :smile: !

**Pull request merged**